modules/dag-shared/src/main/scala/org/tessellation/dag/transaction/TransactionValidator.scala

```diff
@@ -25,13 +25,9 @@ import org.tessellation.security.signature.signature.SignatureProof
 import eu.timepit.refined.auto._
 import io.estatico.newtype.ops._
 
-trait TransactionValidator[F[_]] {
+abstract class TransactionValidator[F[_]: Async: KryoSerializer: SecurityProvider] {
```

Comment:
you can do:

```scala
sealed abstract class TransactionValidator[F[_]: Async: KryoSerializer: SecurityProvider] private (...) {
  ...
}
```

Reply:
Not sure if it makes sense to make the constructor private. I need to be able to implement it in different places (DAG L1, L0) based on the storages created there.